mezzanine/core/models.py

```diff
@@ -524,8 +529,22 @@ def get_content_models(cls):
         return [m for m in apps.get_models()
                 if m is not concrete_model and issubclass(m, concrete_model)]
 
+    def set_content_model(self):
+        """
+        Set content_model to the child class's related name, or None if this is
+        the base class.
+        """
+        is_base_class = (
+            base_concrete_model(ContentTyped, self) == self.__class__)
+        self.content_model = (
+            None if is_base_class else self.get_content_model_name())
+
     def get_content_model(self):
-        return getattr(self, self.content_model, None)
+        """
+        Return content model, or if this is the base class return it.
+        """
+        return (getattr(self, self.content_model) if self.content_model
+                else self.__class__)
```

Comment:
This should be `self` rather than `self.__class__` – `get_content_model()` returns an instance.